ddl/cache.lua

```diff
@@ -0,0 +1,136 @@
+local fiber = require('fiber')
+
+local cache = nil
+
+local CACHE_LOCK_TIMEOUT  = 3
+local SPACE_NAME_IDX      = 1
+local SHARD_FUNC_NAME_IDX = 2
+local SHARD_FUNC_BODY_IDX = 3
+
+-- Function decorator that is used to prevent cache_build() from being
+-- called concurrently by different fibers.
+local function locked(f)
+    return function(...)
+        local ok = cache.lock:put(true, CACHE_LOCK_TIMEOUT)
+
+        if not ok then
+            error("cache lock timeout is exceeded")
+        end
+
+        local status, err = pcall(f, ...)
+        cache.lock:get()
+
+        if not status or err ~= nil then
+            return err
+        end
+    end
+end
+
+-- Build cache.
+--
+-- Cache structure format:
+-- cache = {
+--   spaces = {
+--     'space_name' = {
```

Comment:
Nit: In Lua syntax it would be either `{space_name = <...>}` or `{['space_name'] = <...>}`. Or, if you want `space_name` be like a variable, `{[space_name] = <...>}`. The latter variant is used in the README.md.

Reply:
BTW, I would suggest the following style:

```lua
{
    -- A description here.
    field = <...>,
}
```

Instead of inline comments. We usually use this style and, as for me, it generally looks more accurate.

Reply:
fixed